src/app/core/services/label.service.ts

```diff
@@ -70,36 +95,92 @@ export class LabelService {
   }
 
   /**
-   * Stores the json data from Github api into the list of arrays in this service
-   * @param labels: the json data of the label
+   * Returns an array of Preset Labels.
    */
-  private populateLabelLists(labels: Array<{}>): void {
+  private getRequiredLabels(): Label[] {
+    const requiredLabels: Label[] = [];
 
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
+    for (const category of Object.keys(this.REQUIRED_LABELS)) {
+      for (const labels of Object.values(this.REQUIRED_LABELS[category])) {
+        requiredLabels.push(labels as Label);
       }
+    }
+
+    return requiredLabels;
+  }
+
+  /**
+   * Stores the json data from Github api into the list of arrays in this service
+   * @param labels: the json data of the label
+   * @param expectedLabels: An array of labels that are required by the application.
+   */
+  private storeLabelData(labels: Array<{}>, expectedLabels: Label[]): void {
 
-      this.labelColorMap.set(labelValue, labelColor);
-      const labelList = this.getLabelList(labelType);
+    // Parse Input Data to Label[]
+    const labelData: Label[] = this.parseLabelData(labels);
 
-      if (labelList !== undefined) {
-        labelList.push({labelValue: labelValue, labelColor: labelColor});
+    expectedLabels.forEach(label => {
+      // Checks if remote data set already contains correct label.
+      if (labelData.filter(remoteLabel => remoteLabel.equals(label)).length !== 0) {
+        this.saveLabelData(label);
+        return;
+      }
```

Comment:
If you use the algorithm I suggested, you don't need this branch. First, you search for the weaker condition i.e., name equality. In the results, you search for the stronger equivalence.

Reply:
I'll remove this branch and include the check for the same name but different colour below.